src/number_format.js

```diff
@@ -139,9 +142,14 @@ class NumberFormat extends React.Component {
     }
     else{
       let beforeDecimal = formattedValue, afterDecimal = '';
-      const hasDecimals = formattedValue.indexOf(decimalSeparator) !== -1;
+      const hasDecimals = formattedValue.indexOf(decimalSeparator) !== -1 || decimalPrecision !== false;
```

Comment:
Based on current default value of decimalPrecision, user always have to define decimalPrecision otherwise he will not be able to add it.